src/lib/sticky-header/sticky-header-dir.ts

```diff
@@ -0,0 +1,295 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Component, Directive, Input, Output,
+  OnDestroy, AfterViewInit, ElementRef, Injectable, Optional} from '@angular/core';
+import {Scrollable} from '../core/overlay/scroll/scrollable';
+
+
+/**
+ * 'cdkStickyRegion' is a region contains one sticky header and other
+ * contents that user wants to put under the sticky-header. There only can
+ * be one sticky-header in one 'cdkStickyRegion'.
+ * If a user does not define a 'cdkStickyRegion' for a sticky-header, the direct
+ * parent node of the sticky-header will be set as the 'cdkStickyRegion'.
+ */
+@Directive({
+  selector: '[cdkStickyRegion]',
+})
+export class CdkStickyRegion {
+  constructor(private _element: ElementRef) { }
+
+  getElementRef(): ElementRef {
```

Comment:
Rather than having a method to get this, you can change to constructor to
```ts
constructor(public readonly _elementRef: ElementRef) { }
```
And then reference it directly from sticky-header (we use the leading `_` to denote "internal to Angular Material")